docs/architecture.md

```diff
@@ -8,22 +8,28 @@ Architecturally, an application built with DBOS looks like this:
 
 <img src={require('@site/static/img/architecture/dbos-architecture.png').default} alt="DBOS Architecture" width="750" className="custom-img"/>
 
-You install the DBOS library into your existing application running on your existing servers and infrastructure and connect it to a Postgres database.
-Then, you annotate workflows and steps in your application code.
-The DBOS library **checkpoints** those workflows and steps in Postgres.
-If your program fails, crashes, or is interrupted, DBOS uses those checkpoints to automatically recover your workflows from the last completed step.
+You can integrate DBOS into your existing application running on your existing servers and infrastructure in just three steps:
+
+1. Install the DBOS library into your application.
+2. Connect to any Postgres database.
+3. Annotate workflows and steps directly in your application code.
+
+For more detail on how to add DBOS to your application, check out the language-specific integration guides ([Python](./python/integrating-dbos.md), [TypeScript](./typescript/integrating-dbos.md)).
+
+Once integrated, DBOS automatically checkpoints every workflow and step execution in your application to Postgres.
+When failures occur, whether from crashes, interruptions, or restarts, DBOS uses those checkpoints to recover each of your workflows from the last completed step.
```

Comment:
We can put a forward link here to the `How Workflow Recovery Works` paragraph. This is just a high level overview, for people who want to know the details, we can encourage them to read further.